lib/std/array_list.zig

```diff
@@ -1,3 +1,15 @@
+//! `std.array_list` provides the types `ArrayList`, `ArrayListAligned`, `ArrayListUnmanaged`,
+//! and `ArrayListAlignedUnmanaged` for optionally dynamically allocated and resizable lists.
+//! Whenever a function is referred to as (not) causing illegal behavior,
+//! it is true only under the following conditions:
+//! - None of the input arguments is `undefined` or any other illegal value for the given type.
+//! - The implementation of any allocators used complies with the Allocator interface.
+//! - The struct is in a valid state.
+//! - for an unmanaged list and a function accepting an allocator: the struct was not initialized with `initBuffer`.
+//! For functions in which there is a single source of illegal integer overflow,
+//! it is annotated with a comment.
+//! Functions which check for overflow return `error.OutOfMemory`, as integer overflow on `usize`
+//! can be viewed as address space exhaustion.
```

Comment:
It's more helpful to explicitly document things on the functions and types rather than generalize. I think it is better without this text.